application/src/main/java/org/togetherjava/tjbot/utils/ModAuditLogWriter.java

```diff
@@ -0,0 +1,59 @@
+package org.togetherjava.tjbot.utils;
+
+import net.dv8tion.jda.api.EmbedBuilder;
+import net.dv8tion.jda.api.entities.Guild;
+import net.dv8tion.jda.api.entities.TextChannel;
+import net.dv8tion.jda.api.events.interaction.GenericInteractionCreateEvent;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.togetherjava.tjbot.config.Config;
+
+import javax.annotation.Nonnull;
+import java.util.Optional;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+
+public class ModAuditLogWriter {
+    private static final Logger logger = LoggerFactory.getLogger(ModAuditLogWriter.class);
+
+    private static final Predicate<String> isAuditLogChannelName =
+            Pattern.compile(Config.getInstance().getModAuditLogChannelPattern()).asMatchPredicate();
+    private static final Predicate<TextChannel> isAuditLogChannel =
+            channel -> isAuditLogChannelName.test(channel.getName());
+
+    private ModAuditLogWriter() { throw new IllegalStateException("Utility class"); }
+
+    /**
+     * logs an entry in the mod audit log channel.
+     * 
+     * @param guild the guild. usually use {@link GenericInteractionCreateEvent#getGuild()}
+     *
+     * @param embed the embed that will be sent.
+     */
+    public static void log(@Nonnull Guild guild, @Nonnull EmbedBuilder embed) {
+
+        Optional<TextChannel> auditLogChannel = getModAuditLogChannel(guild);
+        if (auditLogChannel.isEmpty()) {
+            logger.warn(
+                    "Unable to log moderation events, did not find a mod audit log channel matching the configured pattern '{}' for guild '{}'",
+                    Config.getInstance().getModAuditLogChannelPattern(), guild.getName());
+            return;
+        }
+
+        auditLogChannel.get().sendMessageEmbeds(embed.build()).queue();
+    }
+
+
+    /**
+     * stolen from {@link org.togetherjava.tjbot.routines.ModAuditLogRoutine}
+     */
+    private static Optional<TextChannel> getModAuditLogChannel(@NotNull Guild guild) {
+        // Check cache first, then get full list
+        return guild.getTextChannelCache()
+            .stream()
+            .filter(isAuditLogChannel)
+            .findAny()
+            .or(() -> guild.getTextChannels().stream().filter(isAuditLogChannel).findAny());
```

Comment:
What do you mean by "slower"? I made a new method because of 

1. the method in [ModAuditLogRoutine](https://github.com/Together-Java/TJ-Bot/blob/develop/application/src/main/java/org/togetherjava/tjbot/routines/ModAuditLogRoutine.java) is `private` and I didn't want to mess with that,
2. the method in [ModAuditLogRoutine](https://github.com/Together-Java/TJ-Bot/blob/develop/application/src/main/java/org/togetherjava/tjbot/routines/ModAuditLogRoutine.java) is dependent by the constructor and it could be a potential problem.

Reply:
I mean the `or` method call.

Reply:
In fact, it is doing the exact same thing in [ModAuditLogRoutine:332-339](https://github.com/Together-Java/TJ-Bot/blob/develop/application/src/main/java/org/togetherjava/tjbot/routines/ModAuditLogRoutine.java#L332-L339).
as I said in the in-code comment, it checks the cache first then gets the full list.

Reply:
@Zabuzard what do you think?

Reply:
@Tais993 can I mark this as resolved, or am I missing something?

Reply:
Unfortunately I missed that code from the PR.

The getGuilds call returns an unmodifiable list of the cache.
So it's a duplicate, which is also a lot slower.

Reply:
do I need to remove something from my code? I didn't quite understand what do you mean

Reply:
> I'll create an issue for the existing bit.

https://github.com/Together-Java/TJ-Bot/commit/29483e8d3da0fccf50b40d8bdc575e2d1c2f9c86, that merges the duplicate methods takes care of it.